elastic/apm-server
# Update to Go 1.19

## Motivation/summary

Update the minimum Go version to 1.19 in go.mod, and update .go-version and other files with the specific Go patch release. We continue to use a specific patch release to ensure builds are reproducible.

We introduce `script/update_go_version.sh` which reads the major.minor version from `go.mod`, uses `gimme` to identify the most recent patch release for that series, and updates `.go-version` and friends.

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~

## How to test these changes

Run `make package`, and verify that the apm-server binaries and Docker images (one of each) it produces use the exact Go version specified in `.go-version`.

## Related issues

Closes https://github.com/elastic/apm-server/issues/9086
